driver-embedded/src/main/com/mongodb/embedded/client/EmbeddedServer.java

```diff
@@ -16,78 +16,79 @@
-import com.mongodb.connection.Stream;
-import com.mongodb.connection.StreamFactory;
 import com.mongodb.diagnostics.logging.Logger;
 import com.mongodb.diagnostics.logging.Loggers;
 import com.mongodb.event.CommandListener;
-import com.mongodb.internal.connection.Authenticator;
 import com.mongodb.internal.connection.ClusterClock;
 import com.mongodb.internal.connection.ClusterClockAdvancingSessionContext;
 import com.mongodb.internal.connection.CommandHelper;
 import com.mongodb.internal.connection.CommandProtocol;
 import com.mongodb.internal.connection.DefaultServerConnection;
 import com.mongodb.internal.connection.DescriptionHelper;
 import com.mongodb.internal.connection.InternalConnection;
-import com.mongodb.internal.connection.InternalStreamConnection;
-import com.mongodb.internal.connection.InternalStreamConnectionInitializer;
 import com.mongodb.internal.connection.LegacyProtocol;
 import com.mongodb.internal.connection.ProtocolExecutor;
 import com.mongodb.session.SessionContext;
+import com.sun.jna.Pointer;
 import org.bson.BsonDocument;
 import org.bson.BsonInt32;
 
 import java.io.Closeable;
-import java.util.Collections;
+import java.io.File;
 
+import static com.mongodb.assertions.Assertions.isTrue;
 import static com.mongodb.internal.async.ErrorHandlingResultCallback.errorHandlingCallback;
 import static com.mongodb.internal.connection.ClientMetadataHelper.createClientMetadataDocument;
 import static com.mongodb.internal.event.EventListenerHelper.getCommandListener;
+import static java.lang.String.format;
 
 class EmbeddedServer implements Server, Closeable {
-    private static final Logger LOGGER = Loggers.getLogger("connection");
-    private static final MongoDriverInformation MONGO_DRIVER_INFORMATION = MongoDriverInformation.builder()
-            .driverName("embedded").build();
-    private final BsonDocument clientMetadataDocument;
-    private final ClusterId clusterId;
+    private static final Logger LOGGER = Loggers.getLogger("embedded.client");
+    private static final MongoDriverInformation MONGO_DRIVER_INFORMATION = MongoDriverInformation.builder().driverName("embedded").build();
     private final ClusterClock clusterClock;
     private final CommandListener commandListener;
     private final ServerAddress serverAddress;
-    private final EmbeddedDatabase embeddedDatabase;
     private final ServerDescription serverDescription;
+    private final EmbeddedInternalConnectionPool connectionPool;
+    private volatile boolean isClosed;
+    private volatile Pointer databasePointer;
 
     EmbeddedServer(final MongoClientSettings mongoClientSettings) {
-        this.embeddedDatabase = new EmbeddedDatabase(mongoClientSettings);
-        this.clientMetadataDocument = createClientMetadataDocument(mongoClientSettings.getApplicationName(), MONGO_DRIVER_INFORMATION);
-        this.clusterId = new ClusterId();
+        this.databasePointer = createDatabasePointer(mongoClientSettings);
```

Comment:
The need for the EmbeddedDatabase class seemed to disappear, so the EmbeddedServer now creates the database Pointer.